src/decode.rs

```diff
@@ -2550,7 +2526,11 @@ fn decode_b(
                 }
                 NEWMV => {
                     let mut mv1d = mvstack[drl_idx as usize].mv.mv[i];
-                    read_mv_residual(f, ts_c, &mut mv1d, !frame_hdr.force_integer_mv);
+                    read_mv_residual(
+                        ts_c,
+                        &mut mv1d,
+                        frame_hdr.hp as i32 - frame_hdr.force_integer_mv as i32,
```

Comment:
Could you put this in a closure (or var, but with a closure, you only have to repeat it once), since C puts it in a separate var?